### PR TITLE
Tsk-1040: Support draft for DraggableList

### PR DESCRIPTION
**Pull request by mrsadman99**
# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: packages/presentation/src/components/DraggableList.svelte [hunk 1]
```diff
@@ -59,7 +67,7 @@
         handleMove(draggingIndex, toIndex)
         return
       }
-      if (!checkHasRank(objects)) {
+      if (!checkHasRank(objects) || isDraft) {
```
Comment (haiodo) [inline packages/presentation/src/components/DraggableList.svelte]: You still need for proper draft update.
```
const [prev, next] = [
        objects[draggingIndex < toIndex ? toIndex : toIndex - 1],
        objects[draggingIndex < toIndex ? toIndex + 1 : toIndex]
      ]
object.rank = calcRank(prev, next)
```

Comment (mrsadman99) [reply packages/presentation/src/components/DraggableList.svelte]: Done.

**Review by haiodo — APPROVED** [review]

**Comment by haiodo** [comment]
@mrsadman99 Please sign commits.

**Comment by mrsadman99** [comment]
> @mrsadman99 Please sign commits.

Done.

**Pull request merged**